### ISSUE TITLE
[v2] `skaffold render` not working for helm renderer (outputs empty string) IF deploy.helm value exists (like it does when `skaffold fix` run)

### ISSUE THREAD

**Issue by aaron-prindle** [GoogleContainerTools/skaffold#7653]
Currently on V2 attempting to use `skaffold render` for both helm style manifest positions yields empty output (tldr; helm render is broken atm).  Below shows how `v1.39.1` renders a helm project vs how `v2` renders a helm project

```
aprindle@aprindle-ssd ~/skaffold/examples/helm-deployment  [main]$ cat skaffold.yaml
apiVersion: skaffold/v2beta29
kind: Config
build:
  artifacts:
  - image: skaffold-helm
deploy:
  helm:
    releases:
    - name: skaffold-helm
      chartPath: charts

```

```
aprindle@aprindle ~/skaffold/examples/helm-deployment  [main]$ skaffold-v1.39.1 render -a build.artifacts 
---
# Source: skaffold-helm/templates/deployment.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: skaffold-helm
  labels:
    app: skaffold-helm
spec:
  selector:
    matchLabels:
      app: skaffold-helm
  replicas: 2
  template:
    metadata:
      labels:
        app: skaffold-helm
    spec:
      containers:
      - name: skaffold-helm
        image: skaffold-helm:latest
aprindle@aprindle ~/skaffold/examples/helm-deployment  [main]$ skaffold-v2 render -a build.artifacts 

aprindle@aprindle ~/skaffold/examples/helm-deployment  [main]$  # ^^^ it was empty
```

this is technically the correct output (although we have a bug to fix this) for the deploy.helm.releases configuration but the output is zero for the `manifests` style config as well:

```
aprindle@aprindle-ssd ~/skaffold/examples/helm-deployment  [main]$ cat skaffold.yaml 
apiVersion: skaffold/v3alpha1
kind: Config
build:
  artifacts:
  - image: skaffold-helm
manifests:
  helm:
    releases:
    - name: skaffold-helm
      chartPath: charts
deploy:
  helm: {}
```

```
aprindle@aprindle ~/skaffold/examples/helm-deployment  [main]$ skaffold-v2 render -a build.artifacts 

aprindle@aprindle ~/skaffold/examples/helm-deployment  [main]$ # ^^^ still empty
```

**Comment by aaron-prindle** [comment]
Seems the issue is that the way the code currently checks for `legacyHelmDeployer` is the existence of a `deploy.helm` stanza:
```
...
deploy:
  helm: {}
```

  If you modify the `skaffold.yaml` to instead be:
```
apiVersion: skaffold/v3alpha1
kind: Config
build:
  artifacts:
  - image: skaffold-helm
manifests:
  helm:
    releases:
    - name: skaffold-helm
      chartPath: charts
# deploy:
  # helm: {}

```
render works as expected

**Comment by aaron-prindle** [comment]
I think this needs to be modified as currently anyone that uses `skaffold fix` on a helm deployer `skaffold.yaml` will have their `skaffold render` + `skaffold apply` workflow broken if they were using that workflow (skaffold render will be empty, they will need to manually tweak their config to remove the deploy.helm stanza)

**Comment by aaron-prindle** [comment]
Fixed with #7654
